pkg/background/generate/generate.go

```diff
@@ -232,16 +233,35 @@ func (c *GenerateController) applyGenerate(resource unstructured.Unstructured, u
 			logger.V(4).Info("querying all update requests")
 			selector := labels.SelectorFromSet(labels.Set(map[string]string{
 				kyvernov1beta1.URGeneratePolicyLabel:       engineResponse.Policy().GetName(),
-				kyvernov1beta1.URGenerateResourceNameLabel: engineResponse.Resource.GetName(),
 				kyvernov1beta1.URGenerateResourceKindLabel: engineResponse.Resource.GetKind(),
 				kyvernov1beta1.URGenerateResourceNSLabel:   engineResponse.Resource.GetNamespace(),
 			}))
-			urList, err := c.urLister.List(selector)
+			// get update requests that have the resource UID label
+			requirement, err := labels.NewRequirement(kyvernov1beta1.URGenerateResourceUIDLabel, selection.Equals, []string{string(engineResponse.Resource.GetUID())})
+			if err != nil {
+				logger.Error(err, "failed to add the resource UID label")
+			}
+			selectorWithResUID := selector.Add(*requirement)
+			urList, err := c.urLister.List(selectorWithResUID)
 			if err != nil {
 				logger.Error(err, "failed to get update request for the resource", "kind", engineResponse.Resource.GetKind(), "name", engineResponse.Resource.GetName(), "namespace", engineResponse.Resource.GetNamespace())
 				continue
```

Comment:
We need to be consistent when the error occurs, should we resume fetching urs by the resourceName label on errors?

Reply:
Yes, I agree. It will be easier for the user to check which resource name the update requests failed for.

Reply:
Should we remove this `continue` to fall into resourceName listing?